app/experimenter/nimbus-ui/src/components/PageEditBranches/FormBranches/FormScreenshot/index.tsx

```diff
@@ -0,0 +1,170 @@
+          defaultValue={imageProps.defaultValue}
+          render={({ onChange, onBlur, value, name, ref }) => {
+            const {
+              "data-testid": dataTestId,
+              className,
+              isInvalid,
+              isValid,
+            } = imageProps;
+            return (
+              <>
+                <Form.File custom>
+                  <Form.File.Input
+                    {...{
+                      "data-testid": dataTestId,
+                      className: classNames(className, "text-nowrap"),
+                      isInvalid,
+                      isValid,
+                      name,
+                      ref,
+                      onBlur,
+                      accept: IMAGE_UPLOAD_ACCEPT,
+                      onChange: (ev: React.ChangeEvent<HTMLInputElement>) => {
+                        imageProps.onChange?.();
+                        return onChange(ev.target.files?.[0]);
+                      },
+                    }}
+                  />
+                  <Form.File.Label data-testid="upload-label">
+                    {uploadToLabel(value)}
+                  </Form.File.Label>
+                  <FormErrors name="image" />
+                </Form.File>
+                <UploadPreviewImage upload={value} />
+              </>
+            );
+          }}
+        />
+      </Form.Group>
+      <Form.Group as={Col} controlId={`${fieldNamePrefix}-description`}>
+        <Form.Label className="w-100">
+          <Container className="m-0 p-0">
+            <Row>
+              <Col>Description</Col>
+              <Col className="text-right">
+                <Button
+                  data-testid="remove-screenshot"
+                  variant="light"
+                  className="bg-transparent border-0 p-0 m-0"
+                  title="Remove screenshot"
+                  onClick={onRemove}
+                >
+                  <DeleteIcon width="18" height="18" />
+                </Button>
+              </Col>
+            </Row>
+          </Container>
+        </Form.Label>
+        <Form.Control
+          {...{
+            ...descriptionProps,
+            className: classNames(descriptionProps.className, "h-100"),
+          }}
+          as="textarea"
+          rows={3}
+        />
+        <FormErrors name="description" />
+      </Form.Group>
+    </Form.Row>
+  );
+};
+
+function uploadToLabel(upload: Upload | null) {
+  if (upload) {
+    if (typeof upload === "string") {
+      return upload;
+    } else if (typeof upload === "object" && "name" in upload) {
+      return upload.name;
+    }
+  }
+  return ""; // null or :shrug:
+}
+
+const UploadPreviewImage = ({ upload }: { upload: Upload | null }) => {
+  let previewUrl = null;
+  if (upload) {
+    if (typeof upload === "string") {
+      previewUrl = upload;
+    } else if (typeof upload === "object" && "name" in upload) {
+      previewUrl = URL.createObjectURL(upload);
+    }
```

Comment:
How come we have to handle both the string and object case?

Reply:
The string case represents an existing screenshot on the server, where an upload doesn't happen at time of submission and we should set the GQL property to null so as not to replace or delete it.